src/app/clusters/scenes-server/scenes-server.cpp

```diff
@@ -0,0 +1,940 @@
+            ctxt, [this](HandlerContext & ctx, const auto & req) { HandleStoreScene(ctx, req); });
+        return;
+    case Commands::RecallScene::Id:
+        HandleCommand<Commands::RecallScene::DecodableType>(
+            ctxt, [this](HandlerContext & ctx, const auto & req) { HandleRecallScene(ctx, req); });
+        return;
+    case Commands::GetSceneMembership::Id:
+        HandleCommand<Commands::GetSceneMembership::DecodableType>(
+            ctxt, [this](HandlerContext & ctx, const auto & req) { HandleGetSceneMembership(ctx, req); });
+        return;
+    case Commands::EnhancedAddScene::Id:
+        HandleCommand<Commands::EnhancedAddScene::DecodableType>(
+            ctxt, [this](HandlerContext & ctx, const auto & req) { HandleEnhancedAddScene(ctx, req); });
+        return;
+    case Commands::EnhancedViewScene::Id:
+        HandleCommand<Commands::EnhancedViewScene::DecodableType>(
+            ctxt, [this](HandlerContext & ctx, const auto & req) { HandleEnhancedViewScene(ctx, req); });
+        return;
+    case Commands::CopyScene::Id:
+        HandleCommand<Commands::CopyScene::DecodableType>(
+            ctxt, [this](HandlerContext & ctx, const auto & req) { HandleCopyScene(ctx, req); });
+        return;
+    }
+}
+
+CHIP_ERROR ScenesServer::EnumerateAcceptedCommands(const ConcreteClusterPath & cluster, CommandIdCallback callback, void * context)
+{
+    using namespace Clusters::Scenes::Commands;
+
+    constexpr CommandId acceptedCommandsList[] = {
+        AddScene::Id,    ViewScene::Id,          RemoveScene::Id,      RemoveAllScenes::Id,   StoreScene::Id,
+        RecallScene::Id, GetSceneMembership::Id, EnhancedAddScene::Id, EnhancedViewScene::Id, CopyScene::Id,
+    };
+
+    for (const auto & cmd : acceptedCommandsList)
+    {
+        if (callback(cmd, context) != Loop::Continue)
+        {
+            break;
+        }
+    }
+
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR ScenesServer::EnumerateGeneratedCommands(const ConcreteClusterPath & cluster, CommandIdCallback callback, void * context)
+{
+
+    using namespace Clusters::Scenes::Commands;
+
+    constexpr CommandId generatedCommandsList[] = {
+        AddSceneResponse::Id,         ViewSceneResponse::Id,         RemoveSceneResponse::Id,
+        RemoveAllScenesResponse::Id,  StoreSceneResponse::Id,        GetSceneMembershipResponse::Id,
+        EnhancedAddSceneResponse::Id, EnhancedViewSceneResponse::Id, CopySceneResponse::Id,
+    };
+
+    for (const auto & cmd : generatedCommandsList)
+    {
+        if (callback(cmd, context) != Loop::Continue)
+        {
+            break;
+        }
+    }
+    return CHIP_NO_ERROR;
+}
+
+// AttributeAccessInterface
+CHIP_ERROR ScenesServer::Read(const ConcreteReadAttributePath & aPath, AttributeValueEncoder & aEncoder)
+{
+    switch (aPath.mAttributeId)
+    {
+    case Attributes::SceneCount::Id:
+        ReturnErrorOnFailure(mSceneTable->GetGlobalSceneCount(mSceneCount));
```

Comment:
GetSceneTable now requires Endpoint to function (invalid endpoint will be used otherwise, which will result in an error when trying to store or read the nvm)